tests/AppBundle/FakeGithubDownloader.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace Tests\AppBundle;
+
+use AppBundle\GithubDownloaderInterface;
+use Lpdigital\Github\Entity\PullRequest;
+
+class FakeGithubDownloader implements GithubDownloaderInterface
+{
+    public function downloadAndExtract(PullRequest $pullRequest, $head = true): string
+    {
+        $info = true === $head ? $pullRequest->getHead() : $pullRequest->getBase();
+
+        return $pullRequest->getId().'/'.$info['sha'];
```

Comment:
Be careful: sometimes GitHub API return weird answers (I see here you assume `$info` has `sha` property.

For example if you perform GitHub API calls with an un-authenticated token GA might return a 200 HTTP with weird stuff in it instead of returning a 400 or 401. Because it considers you're performing "anonymous" calls.

Reply:
I think we're pretty safe here as `$pullRequest` is an object made from the Github payload and would have thrown an exception if not complete ;)